### DISCUSSION TITLE
The color coding of Contribution Graph is showing wrong Information

### DISCUSSION THREAD

**Discussion by shubham15gupta09** [community/community#23261] [New to GitHub]
<p>As We all know GitHub uses a color coding to show how much you have contributed in a particular day using 5 different shaded starting White to Dark Green .</p>
<p>But from last few days I am seeing that the color coding is not correct according to the contribution i made .</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f094ed30c6d42301a5f14d0dd70f842bd9e1233a.png" data-download-href="/uploads/short-url/ykhFk5HpRtfKwnyDqyhSpwVTDqW.png?dl=1" title="git1.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/37848656/181075956-45243e68-7a4b-4f10-a223-c6bd0a89b4a6.png" alt="git1.png" data-base62-sha1="ykhFk5HpRtfKwnyDqyhSpwVTDqW" width="690" height="155" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f094ed30c6d42301a5f14d0dd70f842bd9e1233a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">git1.png</span><span class="informations">1428×322 12.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>As you can see from the graph I did a lot of commits but some area are still showing Light Green .</p>
<p>Can someone help me ?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/shubham15gupta09">@shubham15gupta09</a>,</p>
<p>The colors on the contribution graph are determined by looking at all of the days currently being displayed, and then allocating shades of green relative to what’s there.</p>
<p>That means that if you have a new day with a record amount of activity, much more than your previous days, that day will become the darkest shade of green, and all of the other days will become lighter.</p>
<p>Similarly, if your busiest day drops off the end of the graph, a new busiest day will be selected, and some squares will become darker.</p>
<p>The cut offs for the different shades of green are calculated separately for each user based on his or her contributions over the last 365 days. We use quartiles after removing outliers. This means that one user might get the darkest shade of green by making 5 commits, while another user will need 50 commits to get the same shade.</p>
<p>I do apologise if I am repeating myself, but I’m afraid we do not discuss the details behind our algorithms as that would allow certain users to game them. I’m afraid we won’t be able to share more than that.</p>

**Reply by kinsDev** [reply]
Actually, this response is enough.

**Reply by ritankarsaha** [reply]
I was facing a similar issue now, thanks it helped me in understanding the situation. This explanation is really quite precise and informative

**Reply by AnkiCodeAdventures** [reply]
i made a contribution of 19 and 20 on just 2 days, and all of my contributions became less intense

**Reply by ritankarsaha** [reply]
> i made a contribution of 19 and 20 on just 2 days, and all of my contributions became less intense

Yes, probably because you made a lot of contributions on the 19th and 20th of July. The graph basically changes depending on your contributions. The day u made the most contributions gets the darkest shade of green and with respect to that other shades of green are marked to the other days.
Contributions can be anything from making commits to opening issues and creating new repos.

**Comment by shubham15gupta09** [comment]
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> </p>
<p>This is something Interesting  :slight_smile:</p>
<p>Does it means I should contribute in a equal manner OR similar amount of contribution everyday .?</p>
<p>Anyway Thanks for your help. </p>

**Reply by Soumilgit** [reply]
I'm having the same doubt.FIY in my profile too it shows lighter green shade as less and darker green shade as more contribution activity,which should ideally be the opposite,despite having around 500 commits.

**Reply by HimaniD27** [reply]
You must be using light themed Github. The colour coding is opposite in light and dark themes. In dark theme, the dark green shade represents less contribution with respect to light green shade.

**Reply by Soumilgit** [reply]
OK fine

**Reply by Soumilgit** [reply]
Will check

**Reply by HimaniD27** [reply]
fine

**Comment by prajvals** [comment]
Hi Andrea, is there any configuration in which we can remove the outlier or just showing in some different color while making the second highest contribution on any day as reference.

Thanks in Advance @AndreaGriffiths11

**Comment by chivicoko** [comment]
if my contribution color turn from green to blue, what does it mean?

**Reply by ghost** [reply]
colour blind

**Reply by codebylanre** [reply]
💀

**Comment by Balcihatice** [comment]
My additive color does not appear green even though I have contributed. how do i straighten

**Comment by Dev-98** [comment]
I think Github updated the color range from ( white - Dark green), I am seeing (black, orange, and yellow)

**Comment by pawan-thakuni-au45** [comment]
i am still getting problem, pushing code everyday but not able to see any contibution there in my profile ,no green color,pleasae help

**Reply by ChingizGodin** [reply]
Check the Contribution settings and ensure that private contributions are turned on. If you're pushing code from a private repository, but private contributions are turned off, your contributions won’t be visible on your profile. Good luck!

**Comment by pawan-thakuni-au45** [comment]
I have resolved this issue, you just need to check your email, is it same
in GitHub and in tool which you are using to push code

**Comment by Soumilgit** [comment]
it's different

**Comment by YassineCherkaoui** [comment]
Hi @shubham15gupta09 ,

Understanding how the colors on the contribution graph are determined can help shed some light on how your activity is represented. The colors are calculated by examining all the days currently displayed on the graph and assigning shades of green relative to the activity present.

For instance, if you have a day with a notably high level of activity, surpassing your previous days, it will be represented as the darkest shade of green. This dynamic allocation of color ensures that your most active days stand out visually.

Conversely, if your most active day drops off the visible range of the graph, a new busiest day will be identified, and the color distribution will adjust accordingly, with some squares becoming lighter.

The cutoffs for the different shades of green are individually calculated for each user based on their contributions over the past 365 days. We utilize quartiles after removing outliers to establish these thresholds. This means that the number of commits required to achieve the darkest shade of green may vary from one user to another.

While I'd love to delve into more specifics, unfortunately, we can't disclose further details about our algorithms. This policy is in place to prevent users from exploiting or manipulating the system. I hope this explanation helps clarify how the contribution graph colors are determined! Let me know if you have any other questions.

**Reply by ebndev** [reply]
Hi @YassineCherkaoui, 

Thanks for wanting to engage in Community Discussion! Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by Soumilgit** [comment]
ok fine

**Comment by ViniciusMarins** [comment]
I think Github updated the color to orange for a while, but now it is green again

**Comment by Soumilgit** [comment]
I use in dark mode only,but now it's rectified👍

**Comment by Ashishkumar667** [comment]
hi @AndreaGriffiths11 ,

Thanks , I have an issue also ....My contribution keeps decreasing, from last two days it was around 476 but now it is 475.It is not the last time, I have faced this even before.Otherwise I would easily crossed 500-510 easily.
